src/rules/pyupgrade/rules/import_replacements_six.rs

```diff
@@ -0,0 +1,200 @@
+    m.insert("response", "urllib.response");
+    m.insert("robotparser", "urllib.robotparser");
+    m
+});
+
+fn refactor_segment(
+    locator: &Locator,
+    stmt: &Stmt,
+    replace: &Lazy<HashMap<&str, &str>>,
+    names: &[Located<AliasData>],
+    module: &str,
+) -> Option<String> {
+    let mut new_entries = String::new();
+    let mut keep_names: Vec<AliasData> = vec![];
+    let mut clean_names: Vec<AliasData> = vec![];
+    for name in names {
+        clean_names.push(name.node.clone());
+    }
+
+    let formatting = ImportFormatting::new(locator, stmt, names);
+    for name in names {
+        let import_name = name.node.name.as_str();
+        match replace.get(import_name) {
+            None => keep_names.push(name.node.clone()),
+            Some(item) => {
+                // Only replace if the name remains the name, or if there is an as on the import
+                let new_name = item.split('.').last().unwrap_or_default();
+                if name.node.asname.is_some() || import_name == new_name {
+                    new_entries.push_str(&format!("{}import {item}", formatting.start_indent));
+                    if let Some(final_name) = &name.node.asname {
+                        new_entries.push_str(&format!(" as {final_name}"));
+                    }
+                    new_entries.push('\n');
+                }
+            }
+        }
+    }
+    // If nothing was different, there is no need to change
+    if new_entries.is_empty() {
+        return None;
+    }
+    let mut final_str = get_fromimport_str(
+        &keep_names,
+        module,
+        formatting.multi_line,
+        &formatting.indent,
+        &formatting.short_indent,
+    );
+    final_str.push_str(&format!("\n{new_entries}"));
+    if final_str.ends_with('\n') {
+        final_str.pop();
+    }
+    Some(final_str)
+}
+
+/// If the entire replace is before the import, we can use this to quickly make
+/// the change
+fn replace_from_only(
+    module: &str,
+    locator: &Locator,
+    stmt: &Stmt,
+    replace: &Lazy<HashMap<&str, &str>>,
+    replace_str: &str,
+) -> Option<String> {
+    let new_module_text = module.replace(&format!("{replace_str}."), "");
+    if let Some(item) = replace.get(new_module_text.as_str()) {
+        let original = locator.slice_source_code_range(&Range::from_located(stmt));
+        let new_str = original.replace(module, item);
+        return Some(new_str);
+    }
+    None
+}
+
+/// UP036
+pub fn import_replacements_six(
+    checker: &mut Checker,
+    stmt: &Stmt,
+    module: &Option<String>,
+    names: &[Located<AliasData>],
+) {
+    // Pyupgrade only works with import_from statements, so this linter does that as
+    // well
+    let final_string: Option<String>;
+    if let Some(module_text) = module {
+        if module_text == "six.moves" {
+            final_string =
+                refactor_segment(checker.locator, stmt, &REPLACE_MODS, names, module_text);
```

Comment:
Pass `checker.stylist` here (charliermarsh prefers just after locator)

Reply:
Do you mind clarifying what you mean here. I interpret it as you want me to use `checker.stylist.locator`, but the `locator` member of `Stylist` is private. So I am assuming you mean something else.